### PR TITLE
run_linux: fix tight loop if file is not pollable

### PR DESCRIPTION
**Pull request by giuseppe**
do not attempt again to poll the file if it is not pollable and the
the errno is not syscall.EINTR or syscall.EAGAIN.

Closes: https://github.com/containers/buildah/issues/2194

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: run_linux.go [hunk 1]
```diff
@@ -1203,6 +1203,13 @@ func runCopyStdio(stdio *sync.WaitGroup, copyPipes bool, stdioPipe [][]int, copy
 	runCopyStdioPassData(copyPipes, stdioPipe, finishCopy, relayMap, relayBuffer, readDesc, writeDesc)
 }
 
+func canRetry(err error) bool {
+	if errno, isErrno := err.(syscall.Errno); isErrno {
```
Comment (rhatdan) [inline run_linux.go]: Should we use an errors.Cause for this?

Comment (giuseppe) [reply run_linux.go]: the error is coming from `unix.Read` and I've verified locally it works.  I can make it safer with errors.Cause if you'd like

**Comment by TomSweeneyRedHat** [comment]
LGTM
FWIW, I'd lean to the errors.Cause too

**Comment by rhatdan** [comment]
@giuseppe Could you rebase and we can merge.  Bors is not happy and I am not sure why.
bors retry

**Pull request merged**